src/cmds/qsub.c

```diff
@@ -966,1202 +764,141 @@ interactive_port(void)
 	return (portstring);
 }
 
-#ifndef WIN32
 /**
  * @brief
- *	This function creates a socket to listen for "X11" data
- *	and returns a port number where its listening for X data.
+ *	Shut and Close a socket
  *
- * @return	char*
- * @retval	portstring	success
+ * @param	sock	file descriptor
  *
- * @par Side Effects
- *		If this function fails, it will exit the qsub process.
+ * @return Void
  *
  */
-static char*
-port_X11(void)
+static void
+shut_close_sock(int sock)
 {
-	pbs_socklen_t namelen;
-	struct sockaddr_in myaddr;
-	static char X11_port_str[X11_PORT_LEN];
-	unsigned short X11_port;
-
-	X11_comm_sock = socket(AF_INET, SOCK_STREAM, 0);
-	if (X11_comm_sock < 0) {
-		perror("qsub: unable to create socket");
-		exit_qsub(1);
-	}
-	myaddr.sin_family = AF_INET;
-	myaddr.sin_addr.s_addr = INADDR_ANY;
-	myaddr.sin_port = 0;
-
-	if (bind(X11_comm_sock, (struct sockaddr *) &myaddr,
-		sizeof(myaddr)) < 0) {
-		perror("qsub: unable to bind to socket");
-		exit_qsub(1);
-	}
-	/* get port number assigned */
-	namelen = sizeof(myaddr);
-	if (getsockname(X11_comm_sock, (struct sockaddr *) &myaddr,
-		&namelen) < 0) {
-		perror("qsub: unable to get port number");
-		exit_qsub(1);
-	}
-	X11_port = ntohs(myaddr.sin_port);
-	(void) sprintf(X11_port_str, "%u", (unsigned int) X11_port);
-	if (listen(X11_comm_sock, 1) < 0) {
-		perror("qsub: listening on X11 socket failed");
-		exit_qsub(1);
-	}
-	return (X11_port_str);
+	shutdown(sock, 2);
+#ifdef WIN32
+	closesocket(sock);
+#else
+	close(sock);
```

Comment:
You wont need this line. see https://github.com/openpbs/openpbs/pull/1917/files#diff-6248e35fa28821a275554b0df12547b1R1413-R1418